include/d/actor/d_a_obj_swheavy.h

```diff
@@ -1,27 +1,53 @@
 #ifndef D_A_OBJ_SWHEAVY_H
 #define D_A_OBJ_SWHEAVY_H
 
-#include "f_op/f_op_actor.h"
+#include "d/d_a_obj.h"
 #include "d/d_bg_w.h"
 
 namespace daObjSwheavy {
     class Act_c : public fopAc_ac_c {
     public:
-        void attr() const {}
-        void is_switch() const {}
-        void off_switch() const {}
-        void on_switch() const {}
-        void prm_get_swSave() const {}
-        void prm_get_type() const {}
-        void rev_switch() const {}
+        struct Attr_c {
+            u8 m00[0x4];
+            int mFlags;
+            float mVSpring;
+            float mVSpeedDecay;
+            float mPushVSpeed0;
+            short m14;
+            short mMiniPushDelay;
+            short mPushDelay;
+        };
+        const Attr_c& attr() const { return M_attr[mType]; }
+
+        enum Flag {
+            FLAG_OBEY_SAVE  = 0x1,
+            FLAG_STAY_PRESSED = 0x2,
+            FLAG_IS_TOGGLE  = 0x4,
+            FLAG_REQ_HEAVY  = 0x8,
+        };
```

Comment:
is it ok/good practice to make enums for stuff like this? are there naming conventions?